### ISSUE TITLE
Pie chart shows spinner, when returned measure/value is `null` or `0`

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#13626]
**Describe the bug**
When a Pie visualization returns `null` or `0` (zero) as the measure/value for all the dimensions, then it causes loading-spinner on dashboard, or overlay with inaccessible error in question.

**To Reproduce**
1. Custom question > Sample Dataset > Orders
2. Custom Column formula `case([ID] = 2, 0, [Discount])` as "NewDiscount"
3. Filter by "ID" is `1, 2, 3`
4. Summarize "Sum of NewDiscount" grouped by "ID"
![image](https://user-images.githubusercontent.com/1447303/97586304-657ad800-19fa-11eb-90d2-7657dcb74caf.png)
5. Change visualization to Pie and save as "Q1"
6. Add "Q1" to dashboard, add filter ID and connect to Order.ID and save dashboard
7. Input `1` or `2` in filter causes loading spinner - and the browser console is full of errors - and then changing the filter to `3`, without doing a browser refresh first, will then cause some funky looking breakage.
![image](https://user-images.githubusercontent.com/1447303/97587575-d373cf00-19fb-11eb-9324-5e3b93a7a9ab.png)
8. Going to question "Q1" and adding another filter "ID"=`1` will cause overlay on top of the error
![image](https://user-images.githubusercontent.com/1447303/97587794-146be380-19fc-11eb-8f57-021a521d6fca.png)

**Expected behavior**
More like 0.36.8, but perhaps with dimensions still shown with zero value?
Doing first part of step 7, or step 8 on 0.36.8 will just show a gray circle without any other details.
![image](https://user-images.githubusercontent.com/1447303/97588160-82180f80-19fc-11eb-8497-df6734570304.png)

**Information about your Metabase Installation:**
Metabase 0.37.0.2

**Additional context**
Related to #12586 and #12600
While it's related to #13571, it wasn't fixed by #13596.

**Comment by nemanjaglumac** [comment]
@flamber - is that an empty string for a custom column name in step `#2`?

**Comment by flamber** [comment]
@nemanjaglumac Someone (not mentioning any names) forgot to copy-pasta correctly. Updated.

**Comment by paulrbr-fl** [comment]
👋 Hi, we experienced this bug since the 0.37 upgrade and it can be quite blocking for Dashboards which have a Pie Chart with empty result in the middle.

Indeed all cards within the dashboards that are presented **after** such problematic Pie Charts are not loaded (all cards have an endless loading spinner).

As a (*pretty hacky*) workaround for us we modified our Pie Chart SQL queries as such:
```sql
WITH question AS (
...<all previous question code>...
)

SELECT * FROM question
UNION
SELECT 
    CASE WHEN SUM(count) = 0 THEN 1 ELSE 0 END,
     '⚠️∅ Empty result (to remove when metabase/metabase#13626 is fixed) ∅⚠️'
FROM question

**Comment by camsaul** [comment]
Fixed by #13721
